eng/scripts/CodeCheck.ps1

```diff
@@ -191,18 +191,25 @@ try {
         }
     }
 
-    Write-Host "Checking for changes to API baseline files"
+    $targetBranch = $env:SYSTEM_PULLREQUEST_TARGETBRANCH
+    $sourceBranch = $env:SYSTEM_PULLREQUEST_SOURCEBRANCH
 
     # Retrieve the set of changed files compared to main
-    $commitSha = git rev-parse HEAD
-    $changedFilesFromMain = git --no-pager diff origin/main...$commitSha --ignore-space-change --name-only --diff-filter=ar
+    Write-Host "Checking for changes to API baseline files $targetBranch...$sourceBranch"
+
+    $changedFilesFromTarget = git --no-pager diff origin/$targetBranch...origin/$sourceBranch --ignore-space-change --name-only --diff-filter=ar
```

Comment:
nit: Why is the source branch useful here❔ Should be on that branch in the working directory.
```suggestion
    $changedFilesFromTarget = git --no-pager diff origin/$targetBranch --ignore-space-change --name-only --diff-filter=ar
```